### PR TITLE
Settings

### PR DESCRIPTION
**Pull request by lagbox**
A persistent settings system.

### CODE REVIEW

**Comment by lagbox** [comment]
I will squash these down when ready.

I need to test this middleware still.

File: app/Data/database/migrations/2016_02_12_070250_create_settings_table_migration.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateSettingsTableMigration extends Migration
```
Comment (lagbox) [inline app/Data/database/migrations/2016_02_12_070250_create_settings_table_migration.php:6]: @Nitpick-CI well thats like your opinion man.

Comment (ryanwinchester) [inline app/Data/database/migrations/2016_02_12_070250_create_settings_table_migration.php:6]: haha

**Comment by ryanwinchester** [comment]
Is this good to go?

**Comment by ryanwinchester** [comment]
Also, I came across this:

https://laravel.com/api/5.1/Illuminate/Foundation/Http/Kernel.html#method_pushMiddleware

```php
public function boot(\Illuminate\Contracts\Http\Kernel $kernel)
{
     $kernel->appendMiddleware('Sheepy85\L5Localization\Middleware\Localization');
}
```

as a way to register the middleware in the ServiceProvider :)

**Comment by ryanwinchester** [comment]
I came across this:

https://laravel.com/api/5.1/Illuminate/Foundation/Http/Kernel.html#method_pushMiddleware

``` php
public function boot(\Illuminate\Contracts\Http\Kernel $kernel)
{
     $kernel->pushMiddleware(\Flashtag\Data\Settings\SettingsMiddleware::class);
}
```

as a way to register the middleware in the ServiceProvider :)

**Comment by lagbox** [comment]
Pushing the middleware works just fine. Tested the middleware as well.

**Pull request merged**